internal/packager/bundle.go

```diff
@@ -0,0 +1,101 @@
+package packager
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/deislabs/cnab-go/bundle"
+	"github.com/docker/app/internal/store"
+	"github.com/docker/app/types"
+	"github.com/docker/app/types/metadata"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/distribution/reference"
+	dockertypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/pkg/jsonmessage"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+func MakeBundleFromApp(dockerCli command.Cli, app *types.App, refOverride reference.NamedTagged) (*bundle.Bundle, error) {
+	logrus.Debug("Making app bundle")
+	meta := app.Metadata()
+	invocationImageName, err := MakeInvocationImageName(meta, refOverride)
+	if err != nil {
+		return nil, err
+	}
+
+	buildContext := bytes.NewBuffer(nil)
+	if err := PackInvocationImageContext(dockerCli, app, buildContext); err != nil {
+		return nil, err
+	}
+
+	logrus.Debugf("Building invocation image %s", invocationImageName)
+	buildResp, err := dockerCli.Client().ImageBuild(context.TODO(), buildContext, dockertypes.ImageBuildOptions{
+		Dockerfile: "Dockerfile",
+		Tags:       []string{invocationImageName},
+		BuildArgs:  map[string]*string{},
+	})
+	if err != nil {
+		return nil, err
+	}
+	defer buildResp.Body.Close()
+
+	if err := jsonmessage.DisplayJSONMessagesStream(buildResp.Body, ioutil.Discard, 0, false, func(jsonmessage.JSONMessage) {}); err != nil {
+		// If the invocation image can't be found we will get an error of the form:
+		// manifest for docker/cnab-app-base:v0.6.0-202-gbaf0b246c7 not found
+		if err.Error() == fmt.Sprintf("manifest for %s not found", BaseInvocationImage(dockerCli)) {
+			return nil, fmt.Errorf("unable to resolve Docker App base image: %s", BaseInvocationImage(dockerCli))
+		}
+		return nil, err
+	}
+
+	return ToCNAB(app, invocationImageName)
+}
+
+func MakeInvocationImageName(meta metadata.AppMetadata, refOverride reference.NamedTagged) (string, error) {
```

Comment:
I wonder why this function are exported?
`MakeInvocationImageName`
`PersistInBundleStore`

Reply:
PersistInBundleStore is used from build/build.go